datafusion/common/src/config.rs

```diff
@@ -1600,13 +1600,13 @@ impl CsvOptions {
     /// Set true to indicate that the first line is a header.
     /// - default to true
     pub fn with_has_header(mut self, has_header: bool) -> Self {
-        self.has_header = has_header;
+        self.has_header = Some(has_header);
         self
     }
 
     /// True if the first line is a header.
     pub fn has_header(&self) -> bool {
```

Comment:
I wonder if it would be clearer to change this to to return `Option<bool>` to mirror the underlying struct 🤔